### DISCUSSION TITLE
Add new methods when(ColumnName) as (whereColumnName) in Eloquent

### DISCUSSION THREAD

**Discussion by rasyidly** [laravel/framework#46991] [Ideas]
- Laravel Version: 10.5.1
- PHP Version: 8.2.4
- Database Driver & Version: MySQL 8.0.31

### Description

I really understand the use of where combined with column names in the database, like the example below:

``` php
User::whereName('John Doe')
```

The query will execute SQL query ``WHERE `name` = 'John Doe'``, so, I have an idea to handle the case if the user searches for User name with an empty value but wants to retrieve all data,

I am trying to use the `scope` and `when` combination in Laravel for the purpose that if the value is empty it will not be processed, I have made a sample below:

``` php
function scopeWhenName($query, $value)
{
     $query->when($value, fn ($subquery) => $subquery->whereName($value));
}
```

The implementation is as follows:

``` php
User::whenName($request->get('name'))->get()
```

The first condition, if the user fills in the `name` query, User will appear with the name that matches the filled query, however, if the query is null/false, then it will skip `WHERE` clauses.

Wouldn't it be nice if `when{ColumnName}` was provided by default by Laravel exactly like the `where{ColumnName}` concept? I think it will be more dynamic for the above case, thanks.

**Comment by dantama** [comment]
I think that method is very nice if added, it will shorten the code quite a bit, and it looks nicer
